pkg/controllerutils/finalizers.go

```diff
@@ -30,6 +30,28 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// PatchFinalizers adds the given finalizers to the object via a patch request.
+func PatchFinalizers(ctx context.Context, c client.Client, obj kutil.Object, finalizers ...string) error {
+	beforePatch := obj.DeepCopyObject()
+
+	f := sets.NewString(obj.GetFinalizers()...)
+	f.Insert(finalizers...)
+	obj.SetFinalizers(f.UnsortedList())
+
+	return c.Patch(ctx, obj, client.MergeFrom(beforePatch))
```

Comment:
Should we rather use a locked patch request?
```suggestion
	return c.Patch(ctx, obj, client.MergeFromWithOptions(beforePatch, client.MergeFromWithOptimisticLock{}))
```

Reply:
What's the benefit?

Reply:
It also sends the `resourceVersion` with the request, just like with an update request, which will be denied by the API server, if the `resourceVersion` was outdated.
I was wondering, if we should do it when sending critical patches (like updating the finalizers), but I am also not so sure about it myself.